pkg/installmanager/installmanager_test.go

```diff
@@ -647,6 +651,55 @@ func TestInstallManagerSSH(t *testing.T) {
 		})
 	}
 }
+func TestInstallManagerSSHKnownHosts(t *testing.T) {
+	apis.AddToScheme(scheme.Scheme)
+
+	tests := []struct {
+		name         string
+		knownHosts   []string
+		expectedFile string
+	}{
+		{
+			name: "single ssh known host",
+			knownHosts: []string{
+				"192.168.86.100 ecdsa-sha2-nistp256 FOOBAR",
+			},
+			expectedFile: `192.168.86.100 ecdsa-sha2-nistp256 FOOBAR`,
+		},
+		{
+			name: "multiple ssh known hosts",
+			knownHosts: []string{
+				"192.168.86.100 ecdsa-sha2-nistp256 FOOBAR",
+				"192.168.86.101 ecdsa-sha2-nistp256 FOOBAR2",
+				"192.168.86.102 ecdsa-sha2-nistp256 FOOBAR3",
+			},
+			expectedFile: `192.168.86.100 ecdsa-sha2-nistp256 FOOBAR
+192.168.86.101 ecdsa-sha2-nistp256 FOOBAR2
+192.168.86.102 ecdsa-sha2-nistp256 FOOBAR3`,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tempDir, err := ioutil.TempDir("", "installmanagersshknownhosts")
+			require.NoError(t, err)
```

Comment:
Add messages to the require and assert calls to aid in readability of test failures.